### PR TITLE
Implement metadata propagation on the workload cluster nodes

### LINKED ISSUE #155: Support node annotation
**Describe the solution you'd like:**

We need to support the annotation of nodes in order to be able to configure Longhorn to use specific disks. Currently CABPR supports only labels for nodes, we need the ability to also set annotations.

**Why do you want this feature:**

This is created from: https://gitlab.com/sylva-projects/sylva-core/-/issues/417

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
kind/feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #155

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: bootstrap/api/v1alpha1/rke2config_types.go [hunk 1]
```diff
@@ -68,6 +68,14 @@ type RKE2AgentConfig struct {
 	//+optional
 	NodeLabels []string `json:"nodeLabels,omitempty"`
 
+	// NodeAnnotations  Aplying annotations on created nodes post bootstrap phase
+	//
```
Comment (alexander-demicev) [inline bootstrap/api/v1alpha1/rke2config_types.go:72]: let's remove an empty line so it doesn't get into CRD scheme

File: bootstrap/config/default/manager_image_patch.yaml [hunk 2]
```diff
@@ -7,5 +7,5 @@ spec:
   template:
     spec:
       containers:
-        - image: ghcr.io/rancher-sandbox/cluster-api-provider-rke2-bootstrap:dev
+        - image: ghcr.io/rancher-sandbox/cluster-api-provider-rke2-bootstrap-amd64:dev
```
Comment (alexander-demicev) [inline bootstrap/config/default/manager_image_patch.yaml:10]: should this change be excluded?

**Review by alexander-demicev — COMMENTED** [review]
lgtm, 2 small comments

**Comment by richardcase** [comment]
@Danil-Grigorev - i think that we should update the existing e2e to ensure that the labels and annotations do get propogated to the nodes.

**Comment by richardcase** [comment]
Also, the DCO check is failing.

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -16,13 +16,14 @@ jobs:
...
+      - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
-          go-version: 1.19
+          go-version: 1.20.7
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3.7.0
         with:
-          version: v1.50.1
+          version: v1.53.3
```
Comment (alexander-demicev) [inline .github/workflows/golangci-lint.yml:26]:
```suggestion
          version: v1.55.1
```

**Review by alexander-demicev — DISMISSED** [review]

**Comment by alexander-demicev** [comment]
We can add e2e tests in a follow-up PR as discussed with @Danil-Grigorev

**Review by alexander-demicev — DISMISSED** [review]

**Review by alexander-demicev — APPROVED** [review]

**Pull request merged**